taskcluster/fxci_config_taskgraph/util/integration.py

```diff
@@ -38,14 +41,32 @@ def find_tasks(decision_index_path: str) -> list[dict[str, Any]]:
     else:
         task_graph = response
 
+    return task_graph.values()
+
+
+def find_tasks(
+    decision_index_path: str,
+    include_attrs: dict[str, list[str]],
+    exclude_attrs: dict[str, list[str]],
+) -> list[dict[str, Any]]:
+    """Find tasks targeted by the Decision task pointed to by `decision_index_path`
+    that match the the included and excluded attributes given.
+    """
     tasks = []
-    for task in task_graph.values():
+
+    for task in _find_tasks(decision_index_path):
         assert isinstance(task, dict)
         attributes = task.get("attributes", {})
-        if attributes.get("unittest_variant") != "os-integration":
-            continue
+        skip = False
+        for attr, values in dict(include_attrs).items():
+            if not any([attributes.get(attr, "").startswith(v) for v in values]):
+                skip = True
+
+        for attr, values in dict(exclude_attrs).items():
+            if any([attributes.get(attr, "").startswith(v) for v in values]):
+                skip = True
```

Comment:
This feels like it should be using `attrmatch`:
https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/util/attributes.py#L9

But I guess that wouldn't support the exclude case? It might be nice for `attrmatch` to support excluding attributes as well (maybe by prepending `!` to the name or something).

But this is also follow-up territory.

Reply:
Actually you could call `attrmatch` twice. Once for `include-attrs`, and a second time for `exclude-attrs` but then negate the return value.

Reply:
TIL about `attrmatch`. Indeed, let's use that instead.

Reply:
Switching to this has highlighted that it also may be nice to support prefixes natively in `attrmatch`; for now I've wrapped that portion in a callable that `attrmatch` can use.

Reply:
Ah, I missed the `startswith`, good call